# HG changeset patch
# User Rik <code@nomad.inbox5.com>
# Date 1272085659 25200
#      Fri Apr 23 22:07:39 2010 -0700
# Node ID 0ecbb17d130ad96a595a1dc0cf6996847d716926
# Parent  23c1910dbd8ed87818c56cc8fc627aa67005f7b6
Clear global variables when 'clear -all' called

Behavior now matches documentation and Matlab

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,12 @@
+2010-04-23  Rik <octave@nomad.inbox5.com>
+
+	* symtab.h: Clear global variables when 'clear -all' called
+
 2010-04-23  John W. Eaton  <jwe@octave.org>
 
 	* DLD-FUNCTIONS/fltk_backend.cc	(OpenGL_fltk::OpenGL_fltk):
 	Explicitly initizlize all data members in the order they are declared.
 
 2010-04-23  John W. Eaton  <jwe@octave.org>
 
 	* txt-eng-ft.cc (ft_render::visit): Declare loop counter size_t,
diff --git a/src/symtab.h b/src/symtab.h
--- a/src/symtab.h
+++ b/src/symtab.h
@@ -1275,16 +1275,18 @@ public:
   static void clear (const std::string& name)
   {
     clear_variable (name);
   }
 
   static void clear_all (void)
   {
     clear_variables ();
+    
+    clear_global_pattern ("*");
 
     clear_functions ();
   }
 
   static void clear_variables (scope_id scope)
   {
     symbol_table *inst = get_instance (scope);
 
